From c214769ae96ae9f05f3273fe1cde8194fb66576d Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:09 -0400
Subject: [tools] perf/evsel: Add missing decrement in id sample parsing

Message-id: <1408462094-14194-201-git-send-email-jolsa@redhat.com>
Patchwork-id: 88178
O-Subject: [PATCH RHEL7.1 BZ1131394 200/325] perf evsel: Add missing decrement in id sample parsing
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit dd44bc6be05e4a948124053c8105cfa581177554
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Oct 18 15:29:01 2013 +0300

description
===========
The final array decrement in id sample parsing is missing, which may
trip up the next person adding a sample format, so add it in.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/evsel.c b/tools/perf/util/evsel.c
index 83a8be2..3a334f0 100644
--- a/tools/perf/util/evsel.c
+++ b/tools/perf/util/evsel.c
@@ -1217,6 +1217,7 @@ static int perf_evsel__parse_id_sample(const struct perf_evsel *evsel,
 
   sample->pid = u.val32[0];
   sample->tid = u.val32[1];
+  array--;
  }
 
  return 0;
-- 
1.7.1